### PR TITLE
Enabled testrepo-based tests on Windows

### PR DESCRIPTION
**Pull request by mih**
...as much as feels feasible right now. In any case, this is a huge step forward in terms of critical test coverage on windows. We are now in the position to investigate individual tests, instead of facing global failure.

- [x] cross-platform implementation of get_local_file_url()
- [x] basic testrepo setup is running on windows
- [x] all new test failures due to testrepos being enabled on windows are marked "known" and annotated with a traceback log link
- [x] addurl is not working on appveyor, fix or disable testrepos on appveyor (leaning towards the latter)

There is still an open issue with `get_local_file_ur l()` https://github.com/datalad/datalad/issues/3977, but given that this is needed for #3966 it will have to come in a separate PR.

### CODE REVIEW

File: datalad/support/network.py [hunk 1]
```diff
@@ -900,22 +901,24 @@ def is_ssh(ri):
         or (isinstance(_ri, URL) and _ri.scheme == 'ssh')
 
 
-def get_local_file_url(fname):
+def get_local_file_url(path):
```
Comment (kyleam) [inline datalad/support/network.py]: Grepping around the extensions, I don't see any calls to `get_local_file_url` that use `fname` as a keyword argument, so changing the parameter name should be fine.  But at the same time, any subjective benefit of the name change doesn't seem to be worth making a breaking change.

If we do stick with name change, the docstring needs to be updated.

Comment (mih) [reply datalad/support/network.py]: You are right, I will revert it.

Comment (mih) [reply datalad/support/network.py]: Done

**Comment by mih** [comment]
Given that this PR makes a number of additional tests run, with relatively minimal impact on behavior on other platforms I'll merge this one now. A follow-up PR will (have to) come with #3977, so potential fixups can be incorporated there.

**Pull request merged**